pkg/controller/knativeserving/knativeserving_controller.go

```diff
@@ -259,22 +261,45 @@ func (r *ReconcileKnativeServing) checkDeployments(instance *servingv1alpha1.Kna
 // Check for all dependencies
 func (r *ReconcileKnativeServing) checkDependencies(instance *servingv1alpha1.KnativeServing) error {
 	defer r.updateStatus(instance)
-
-	istio := schema.GroupVersionKind{Group: "networking.istio.io", Version: "v1alpha3", Kind: "virtualservice"}
-	list := &unstructured.UnstructuredList{}
-	list.SetGroupVersionKind(istio)
-	if err := r.client.List(context.TODO(), nil, list); err != nil {
-		msg := fmt.Sprintf("Istio not detected; please install ServiceMesh")
-		instance.Status.MarkDependencyMissing(msg)
-		log.Error(err, msg)
-		return err
+	configuredMembersData := map[string]string{}
+	smmr := &maistrav1.ServiceMeshMemberRoll{}
+	resource := &unstructured.Unstructured{}
+	resource.SetNamespace("istio-system")
+	resource.SetName("default")
+	resource.SetAPIVersion("maistra.io/v1")
+	resource.SetKind("ServiceMeshMemberRoll")
+	smmrData, err := r.config.Get(resource)
+	if smmrData == nil {
+		if err == nil {
+			err = errors.New("serviceMeshMemberRoll does not exist in istio-system namespace")
+		}
+		return checkError(err, instance)
+	}
+	byteData, err := smmrData.MarshalJSON()
+	if err != nil {
+		return checkError(err, instance)
+	}
+	if err = json.Unmarshal(byteData, smmr); err != nil {
+		return checkError(err, instance)
+	}
+	for i, _ := range smmr.Status.ConfiguredMembers {
+		configuredMembersData[smmr.Status.ConfiguredMembers[i]] = smmr.Status.ConfiguredMembers[i]
+	}
+	if _, ok := configuredMembersData[instance.GetNamespace()]; !ok {
+		return checkError(errors.New("knative-serving namespace is not a configured member in serviceMeshMemberRoll"), instance)
```

Comment:
Rather than use the map, I think you can just use maybe a `found` var or possibly look at the nested `available` function in `checkDeployments` for a similar approach.

Reply:
Modified